app/controllers/application_controller.rb

```diff
@@ -113,7 +113,7 @@ def permitted_params
   end
 
   def search_options_from_params(*param_keys)
-    h = permitted_params.slice(*param_keys)
+    h = permitted_params.to_h.slice(*param_keys)
```

Comment:
We can kill these two birds with one stone by converting the params to a hash in the `permitted_params` method itself:
```ruby
  def permitted_params
    @permitted_params ||= params.permit(*PERMITTED_PARAM_KEYS).to_h
  end
```